File: solutions/java/hamming/2/src/main/java/Hamming.java
public class Hamming {
    private int hammingDistance;
    
    
    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) throw new IllegalArgumentException("strands must be of equal length");
        int distance = 0;
        for (int i = 0; i < leftStrand.length(); i++) {
             if (leftStrand.charAt(i) != rightStrand.charAt(i)) distance++;   
        }
        this.hammingDistance = distance;
    }

    public int getHammingDistance() {
        return this.hammingDistance;
    }
}

File: solutions/lua/hamming/2/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    assert(#a == #b, 'strands must be of equal length')
    distance = 0
    for index = 1, #a do
        if a:sub(index, index) ~= b:sub(index, index) then distance = distance + 1 end
    end
    return distance
end

return Hamming